src/ssl.mli

```diff
@@ -471,51 +445,96 @@ val set_ip : socket -> string -> unit
     [select]ing on it; you should not write or read on it. *)
 val file_descr_of_socket : socket -> Unix.file_descr
 
-(** [read sock buf off len] receives data from a connected SSL socket. *)
-val read : socket -> Bytes.t -> int -> int -> int
+(** {2 I/O on SSL sockets} *)
+
+(** The main communication functions below comes with
+    - a "Acquire" version that acquire the runtime while the operation takes place.
+    - a "NoAcquire" version that does not acquire the runtime while the operation takes place.
+
+    They come in two separate module with the same interface.
+
+    The acquire version are recommanded in one case:
+    - A program using several OCaml thread, if the socket is in blocking
+      more. In this case, as the global lock is released, the other OCaml
+      threads may run.
+
+    The No-acquire version is recommanded otherwise:
+    - i.e. the program is purely sequential (one thread)
+    - the socket is non blocking
+
+    Remark: whatever function you use with non-blocking socket, you must
+    capture all exception that requires you to retry later the same operation.
+*)
+
+module type SslCom = sig
```

Comment:
we don't need to export the module type in the signature. Perhaps it'd be better to put it in a `ssl_intf.ml` and refer to it here.

Reply:
I like to have documlentation in the same .mli file as a .ml. This is not a big mater if it is duplicated in both ssl.ml and ssl.mli ?

Reply:
We're exposing a module type that has no use. I'd rather keep the API surface area as small as possible.

Reply:
The module is used, twice:
```
module SslRelease : SslCom
module SslNoRelease : SslCom
```

And is the only place that gives the type of the main ssl function like `connect`, `read`, `ẁrite`, .... They are available directly as before via the include of SslRelease, I do not see how to remove `SslCom`. Or maybe I don't get fully what you mean ?

Reply:
Right, but it doesn't need to be exposed as a `module type`, creating yet another item in the public API. you can put the module type in a `ssl_intf.ml` file and reference it in the signature, for example. I think this talks about it https://www.craigfe.io/posts/the-intf-trick